src/module/system/text-editor.ts

```diff
@@ -1013,14 +1015,17 @@ async function augmentInlineDamageRoll(
     }
 }
 
+function resolveRollData(rollData: EnrichmentOptionsPF2e["rollData"] = {}): RollDataContextPF2e {
+    return rollData instanceof Function ? rollData() : rollData;
+}
+
 interface EnrichmentOptionsPF2e extends EnrichmentOptions {
-    /** @todo as of v13 this should also support functions that produce roll data */
-    rollData?: RollDataPF2e;
+    rollData?: RollDataContextPF2e | (() => RollDataContextPF2e);
     /** Whether to run the enriched string through `UserVisibility.process` */
     processVisibility?: boolean;
 }
 
-interface RollDataPF2e {
+interface RollDataContextPF2e {
```

Comment:
Why the rename here?

Reply:
See PR description. There already is an exported `RollDataPF2e` in `system/rolls.ts`. We could keep both but I thought it'd be safest to rename one.

Reply:
Given that the one in system/rolls extends dice.RollOptions, I think that one might be the one that needs renaming. The RollData here is actually called RollData within foundry itself.

Reply:
Yeah makes sense. I've renamed `RollDataPF2e` in `system/rolls.ts` to `BaseRollDataPF2e` and changed the one in `text-editor` back to `RollDataPF2e`.